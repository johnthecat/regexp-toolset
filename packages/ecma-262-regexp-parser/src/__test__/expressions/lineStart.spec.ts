import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Line Start', () => {
  it('should correctly parse line start in the start of regexp', () => {
    expect(parseRegexp(/^a/)).toMatchSnapshot();
  });
  it('should correctly parse line start in the middle of regexp', () => {
    expect(parseRegexp(/a^a/)).toMatchSnapshot();
  });
  it('should correctly parse line start in the end of regexp', () => {
    expect(parseRegexp(/a^/)).toMatchSnapshot();
  });
});
