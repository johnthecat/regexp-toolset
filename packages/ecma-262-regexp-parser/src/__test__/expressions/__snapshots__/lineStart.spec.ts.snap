// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Line Start > should correctly parse line start in the end of regexp 1`] = `
{
  "body": {
    "end": 2,
    "expressions": [
      {
        "charCode": 97,
        "end": 1,
        "kind": "Char",
        "start": 1,
        "type": "Simple",
        "value": "a",
      },
      {
        "end": 2,
        "kind": "LineStart",
        "start": 2,
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 3,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Line Start > should correctly parse line start in the middle of regexp 1`] = `
{
  "body": {
    "end": 3,
    "expressions": [
      {
        "charCode": 97,
        "end": 1,
        "kind": "Char",
        "start": 1,
        "type": "Simple",
        "value": "a",
      },
      {
        "end": 2,
        "kind": "LineStart",
        "start": 2,
      },
      {
        "charCode": 97,
        "end": 3,
        "kind": "Char",
        "start": 3,
        "type": "Simple",
        "value": "a",
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 4,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;

exports[`Line Start > should correctly parse line start in the start of regexp 1`] = `
{
  "body": {
    "end": 2,
    "expressions": [
      {
        "end": 1,
        "kind": "LineStart",
        "start": 1,
      },
      {
        "charCode": 97,
        "end": 2,
        "kind": "Char",
        "start": 2,
        "type": "Simple",
        "value": "a",
      },
    ],
    "kind": "Alternative",
    "start": 1,
  },
  "end": 3,
  "flags": "",
  "kind": "Regexp",
  "start": 0,
}
`;
